### PR TITLE
fix: webview zoom level persistence on navigation

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Closes https://github.com/electron/electron/issues/40354

Fixes an issue where webview zoom reset on navigation after #39428. This brings behavior back in line with previously expected behavior. The webview zoom level should persist per-session for the same host even if navigations occur, as was the initial intention in https://github.com/electron/electron/pull/8537.

Confirmed to fix https://github.com/ferdium/ferdium-app/issues/1419 with a local build.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: Fixed an issue where zoom level settings did not persist per-session for webviews.

### CODE REVIEW

File: shell/browser/web_contents_zoom_controller.cc [hunk 1]
```diff
@@ -268,11 +269,18 @@ void WebContentsZoomController::ResetZoomModeOnNavigationIfNeeded(
   double old_zoom_level = zoom_map->GetZoomLevel(web_contents());
   double new_zoom_level = zoom_map->GetZoomLevelForHostAndScheme(
       url.scheme(), net::GetHostOrSpecFromURL(url));
-  event_data_ = std::make_unique<ZoomChangedEventData>(
+
+  ZoomChangedEventData zoom_change_data(
       web_contents(), old_zoom_level, new_zoom_level, false, ZOOM_MODE_DEFAULT);
+  for (auto& observer : observers_) {
+    observer.OnZoomChanged(zoom_change_data);
```
Comment (deepak1556) [inline shell/browser/web_contents_zoom_controller.cc:276]: What is the reason for this change ? Shouldn't the `updateState` in `DidFinishNavigation` take care of calling the observers ?

Comment (codebytere) [reply shell/browser/web_contents_zoom_controller.cc:276]: oops, sorry, this should have been deleted!

Comment (codebytere) [reply shell/browser/web_contents_zoom_controller.cc:276]: @deepak1556 that's what this fix is - removing `UpdateState`. It's what causes this bug.

Comment (deepak1556) [reply shell/browser/web_contents_zoom_controller.cc:276]: Can you please expand a bit more on the change. I am failing to understand how the call to observers here is different from `updateState` making the same call at the [end of the navigation cycle](https://github.com/electron/electron/blob/96f5f89f1bcb3aecb6786339cee92e26e651f6ba/shell/browser/web_contents_zoom_controller.cc#L390-L397). Is there a zoom level change in `SetZoomFactorOnNavigationIfNeeded` between these callsites which prompted this change ?

Comment (codebytere) [reply shell/browser/web_contents_zoom_controller.cc:276]: @deepak1556 the codepath you linked in DidFinishNavigation is only hit in the case that `! navigation_handle->IsSameDocument` - the call to UpdateState i removed was happening on all navigations and that was the issue

Comment (codebytere) [reply shell/browser/web_contents_zoom_controller.cc]: @deepak1556 done!

**Comment by deepak1556** [comment]
LGTM, but the failing test seems to be related.

```
Error occurred in handler for 'GUEST_VIEW_MANAGER_CALL': Error: ERR_ABORTED (-3) loading 'app://host1/'
    at rejectAndCleanup (node:electron/js2c/browser_init:4234:25)
    at WebContents.navigationListener (node:electron/js2c/browser_init:4261:28)
    at WebContents.emit (node:events:526:35) {
  errno: -3,
  code: 'ERR_ABORTED',
  url: 'app://host1/'
}
```

the navigation failure seems to trigger the global `unhandledRejection` handler https://github.com/electron/electron/blob/344b7f0d068ae7d20fdebafc2ebadcdc0b4f7f68/spec/node-spec.ts#L876

**Comment by deepak1556** [comment]
I wrote a really bad test in https://github.com/electron/electron/commit/7cc780d077d1aa950b93db7954d1e321ac88e7f6#diff-a29be2762432f6a7fa49cbbf00bfe98a25b91fbac9d540cae2096ac4dc0b3c0d, we shouldn't have process level handler in that spec.
